File: engine/engine-model/src/main/scala/net/server/channel/packet/guild/ChangeGuildEmblemPacket.scala
package net.server.channel.packet.guild

class ChangeGuildEmblemPacket(private var _type: Byte, private var _background: Short, private var _backgroundColor: Byte, private var _logo: Short, private var _logoColor: Byte) extends BaseGuildOperationPacket(_type = _type) {
  def background: Short = _background

  def backgroundColor: Byte = _backgroundColor

  def logo: Short = _logo

  def logoColor: Byte = _logoColor
}

File: engine/engine-model/src/main/java/net/server/channel/packet/guild/ChangeGuildEmblemPacket.java
package net.server.channel.packet.guild;

public class ChangeGuildEmblemPacket extends BaseGuildOperationPacket {
   private final Short background;

   private final Byte backgroundColor;

   private final Short logo;

   private final Byte logoColor;

   public ChangeGuildEmblemPacket(Byte theType, Short background, Byte backgroundColor, Short logo, Byte logoColor) {
      super(theType);
      this.background = background;
      this.backgroundColor = backgroundColor;
      this.logo = logo;
      this.logoColor = logoColor;
   }

   public Short background() {
      return background;
   }

   public Byte backgroundColor() {
      return backgroundColor;
   }

   public Short logo() {
      return logo;
   }

   public Byte logoColor() {
      return logoColor;
   }
}
